config/tests/src/test/java/org/jboss/windup/test/utils/EnumeratedRuleProviderFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.jboss.windup.test.utils;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.jboss.forge.furnace.proxy.Proxies;
+import org.jboss.forge.furnace.util.Predicate;
+import org.jboss.windup.config.WindupRuleProvider;
+
+/**
+ *
+ *  @author Ondrej Zizka, ozizka at redhat.com
+ */
+public class EnumeratedRuleProviderFilter implements Predicate<WindupRuleProvider>
+{
+
+    // Using strings to prevent inequality of same class from different classloaders.
+    private final Set<String> ruleProviders;
+
+
+    public EnumeratedRuleProviderFilter(Class<? extends WindupRuleProvider>... ruleProviders)
+    {
+        this.ruleProviders = new HashSet(ruleProviders.length);
+        for( Class<? extends WindupRuleProvider> ruleProvider : ruleProviders ){
+            if(Proxies.isProxyType(ruleProvider))
+                ruleProvider = (Class<? extends WindupRuleProvider>) Proxies.unwrapProxyTypes(ruleProvider.getClass());
+            this.ruleProviders.add(ruleProvider.getName());
+        }
+    }
+
+
+    @Override
+    public boolean accept(WindupRuleProvider ruleProvider)
+    {
+        Class<? extends WindupRuleProvider> cls = ruleProvider.getClass();
```

Comment:
Again, cls is very commonly used variable name for class, see e.g. half of the AS 7. Changed.